mender-update/daemon/states.cpp

```diff
@@ -401,13 +407,33 @@ void SendStatusUpdateState::DoStatusUpdate(Context &ctx, sm::EventPoster<StateEv
 		}
 	}
 
+	// Push status.
 	auto err = ctx.deployment_client->PushStatus(
 		ctx.deployment.state_data->update_info.id,
 		status,
 		"",
 		ctx.mender_context.GetConfig().server_url,
 		ctx.http_client,
-		result_handler);
+		[result_handler, &ctx](error::Error err) {
+			// If the deployment succeeded, we don't submit logs, so there's nothing
+			// more to do.
+			if (err != error::NoError || !ctx.deployment.failed) {
```

Comment:
:thinking: Shouldn't this be:
```suggestion
			if (err == error::NoError && !ctx.deployment.failed) {
```
?

Reply:
Maybe the comment is a bit misleading, how is this clarification:

> If there is an error, we don't submit logs now, but call the handler, which may schedule a retry later. If there is no error, and the deployment as a whole was successful, then also call the handler here, since we don't need to submit logs at all then.

Reply:
It is much better. Please update the comment in the code :)